ui/src/main/resources/PhenoTips/SimilarCases.xml

```diff
@@ -39,56 +39,53 @@
   <hidden>false</hidden>
   <content>{{velocity}}
 (% class="" id="similar-cases-container"%)(((##
-#if ($xcontext.action == 'view' &amp;&amp; $xwiki.hasAccessLevel('view', $xcontext.user, 'PhenoTips.SimilarCases'))
-  #if (!$services.patientconsents.hasConsent($doc.getName(), "real"))
-    #break
+##
+#macro (__generateRefreshMatchesSelector )
+{{html clean=false}}##
+&lt;div class="refresh-matches-container"&gt;
+&lt;span class="buttonwrapper similarity-results-refresh" id="similarity-results-refresh"&gt;
+  &lt;button id="refresh-all-button"&gt;
+    &lt;span class="fa fa-refresh" title=""&gt;&lt;/span&gt;
+    &lt;span class="refresh-matches-button"&gt;$escapetool.xml($services.localization.render('phenotips.similarCases.refreshMatches.allServers.label'))&lt;/span&gt;
+  &lt;/button&gt;
+  &lt;button id="refresh-remote-select" class="refresh-remote-select"&gt; &lt;span class="fa fa-caret-down" title=""&gt;&lt;/span&gt;&lt;/button&gt;
+  &lt;div id="remote-select-dropdown" class="dropdown-menu"&gt;
+    &lt;a class="local" href="#"&gt;$escapetool.xml($services.localization.render('phenotips.similarCases.refreshMatches.localServer.label'))&lt;/a&gt;
+##=======================================
+## Get list of remote configurations and
+## only display those which have a non-empty token and the checkbox to show in UI is checked
+##=======================================
+#set ($preferenceDoc = $xwiki.getDocumentAsAuthor("XWiki.XWikiPreferences"))
+#set ($allRemotes = $preferenceDoc.getObjects("PhenomeCentral.RemoteMatchingServiceConfiguration"))
+#foreach ($config in $allRemotes)
+  #if ($config.getProperty('remoteAuthToken').getValue() != "" &amp;&amp; $config.getProperty('serverId').getValue() != "" &amp;&amp; $config.getProperty('includeInUI').getValue() == 1)
+    #set ($serverID = $config.getProperty('serverId').getValue())
+    #set ($serverName = $config.getProperty('humanReadableName').getValue())
+    #if ($serverName == "")
+      #set ($serverName = $serverID)
+    #end
+    &lt;a class="${escapetool.xml($serverID)}" href="#"&gt;${escapetool.xml($serverName)}&lt;/a&gt;
```

Comment:
Not sure which line to comment, but IMHO the dropdown for refreshing matches is a bit unintuitive, with the arrow and text that we have now

Reply:
I don't like the way it looks either. It was supposed to mimic the [Bootstrap dropdown buttons](https://getbootstrap.com/docs/4.0/components/dropdowns/) which people are used to, but the styling is too different. I would:
* increase the padding to .3em
* stick the two components (label and arrow) together
* get rid of the icon
* make sure the dropdown actually looks attaced to the button
* align the dropdown text to the right, and make sure it aligns with the button text

![image](https://user-images.githubusercontent.com/651980/62717036-231e4300-b9d1-11e9-9803-7901374690d2.png)